src/optionstr.c

```diff
@@ -1635,6 +1635,62 @@ expand_set_completeopt(optexpand_T *args, int *numMatches, char_u ***matches)
 	    matches);
 }
 
+/*
+ * The 'completeitemalign' option is changed.
+ */
+    char *
+did_set_completeitemalign(optset_T *args UNUSED)
+{
+    char_u	*p = p_cia;
+    unsigned	new_cia_flags = 0;
+    int		seen[3] = { FALSE, FALSE, FALSE };
+    int		count = 0;
+
+    while (*p != NUL)
+    {
+	if (count >= 3)
+	    return e_invalid_argument;
+
+	if (STRNCMP(p, "abbr", 4) == 0)
+	{
+	    if (seen[CPT_ABBR])
+		return e_invalid_argument;
+	    new_cia_flags = new_cia_flags * 10 + CPT_ABBR;
+	    seen[CPT_ABBR] = TRUE;
+	    count++;
+	}
+	else if (STRNCMP(p, "kind", 4) == 0)
+	{
+	    if (seen[CPT_KIND])
+		return e_invalid_argument;
+	    new_cia_flags = new_cia_flags * 10 + CPT_KIND;
+	    seen[CPT_KIND] = TRUE;
+	    count++;
+	}
+	else if (STRNCMP(p, "menu", 4) == 0)
+	{
+	    if (seen[CPT_MENU])
+		return e_invalid_argument;
+	    new_cia_flags = new_cia_flags * 10 + CPT_MENU;
+	    seen[CPT_MENU] = TRUE;
+	    count++;
+	}
+	else
+	    return e_invalid_argument;
+
+        p = vim_strchr(p, ',');
+	if (p == NULL)
+	    break;
+        if (*p == ',')
+            p++;
```

Comment:
This seems to allow invalid option values like `abbr1234,kind,menu`

Reply:
`  if (STRNCMP(p, "abbr", 4) == 0 && (p[4] == ',' || p[4] == NUL))` something like this will avoid invalid option ..any better idea ?

Reply:
oh copy_option_part